src/ui/settings/controls-settings-ui-handler.ts

```diff
@@ -1,691 +1,246 @@
+    for (const icon of this.mappingIcons) {
+      icon.setTexture(config.padType);
+    }
 
-    return true; // Return true to indicate the scroll cursor was successfully updated.
+    this.displaySettingsOptions();
+    if (this.active) {
+      this.updateInstructionIcons();
+    }
   }
 
-  /**
-   * Set the option cursor to the specified position.
-   *
-   * @param settingIndex - The index of the setting.
-   * @param cursor - The cursor position to set.
-   * @param save - Whether to save the setting to local storage.
-   * @returns `true` if the option cursor was set successfully.
-   */
-  public setOptionCursor(settingIndex: number, cursor: number, save?: boolean): boolean {
-    // Retrieve the specific setting using the settingIndex from the settingDevice enumeration.
-    const setting = this.setting[Object.keys(this.setting)[settingIndex]];
-
-    // Get the current cursor position for this setting.
-    const lastCursor = this.optionCursors[settingIndex];
-
-    // Check if the setting is not part of the bindings (i.e., it's a regular setting).
-    if (!this.bindingSettings.includes(setting) && !setting.includes("BUTTON_")) {
-      // Get the label of the last selected option and revert its color to the default.
-      const lastValueLabel =
-        this.optionValueLabels[settingIndex][lastCursor] ?? this.optionValueLabels[settingIndex][0];
-      setTextColor(lastValueLabel, TextStyle.WINDOW);
-
-      // Update the cursor for the setting to the new position.
-      this.optionCursors[settingIndex] = cursor;
-
-      // Change the color of the new selected option to indicate it's selected.
-      const newValueLabel = this.optionValueLabels[settingIndex][cursor] ?? this.optionValueLabels[settingIndex][0];
-      setTextColor(newValueLabel, TextStyle.SETTINGS_SELECTED);
+  protected override setScrollCursor(scrollCursor: number): boolean {
+    const result = super.setScrollCursor(scrollCursor);
+    if (this.mappingValuesText && result) {
+      return this.mappingValuesText.setCursor(scrollCursor);
     }
+    return result;
+  }
 
-    // If the save flag is set, save the setting to local storage
-    if (save) {
-      this.setSetting(setting, cursor);
-    }
+  protected override setTotalRows(rows: number): void {
+    super.setTotalRows(rows);
 
-    return true; // Return true to indicate the cursor was successfully updated.
+    if (this.mappingValuesText) {
+      this.mappingValuesText.setMaxLines(this.rowsToDisplay, true);
+    }
   }
 
-  /**
-   * Update the scroll position of the settings UI.
-   */
-  private updateSettingsScroll(): void {
-    // Return immediately if the options container is not initialized.
-    if (!this.optionsContainer) {
-      return;
+  public override processInput(button: Button): boolean {
+    if (
+      this.noDeviceText.visible
+      && button !== Button.CYCLE_SHINY
+      && button !== Button.CYCLE_FORM
+      && button !== Button.CANCEL
+    ) {
```

Comment:
```suggestion
    const specialButtons: Button[] = [Button.CYCLE_SHINY, Button.CYCLE_FORM, Button.CANCEL];
    if (this.noDeviceText.visible && !specialButtons.includes(button)) {
```
Not sure what to call the `const`.

Reply:
I didn't realize that simply declaring it as a Button array would solve my compilation issues with this approach. thanks :')
I think `navigationButtons` should do the trick to convey why these buttons are not being blocked 🤔 
```suggestion
    const navigationButtons: Button[] = [Button.CYCLE_SHINY, Button.CYCLE_FORM, Button.CANCEL];
    if (this.noDeviceText.visible && !navigationButtons.includes(button)) {
```